### PR TITLE
Add SI prefix

### PR DESCRIPTION
**Pull request by younies**

### CODE REVIEW

File: provider/datagen/src/transform/cldr/units/prefix.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+enum SIPrefix {
+    YOTTA = Prefix { base: 10, power: 24, prefix: "yotta" },
+    ZETTA = Prefix { base: 10, power: 21, prefix: "zetta" },
+    EXA = Prefix { base: 10, power: 18, prefix: "exa" },
+    PETA = Prefix { base: 10, power: 15, prefix: "peta" },
+    TERA = Prefix { base: 10, power: 12, prefix: "tera" },
+    GIGA = Prefix { base: 10, power: 9, prefix: "giga" },
+    MEGA = Prefix { base: 10, power: 6, prefix: "mega" },
```
Comment (younies) [inline provider/datagen/src/transform/cldr/units/prefix.rs:19]: Make it to methods for analyzing the string.

Comment (robertbastian) [reply provider/datagen/src/transform/cldr/units/prefix.rs:19]: `parse_prefix(s: &str) -> Option<(u8, u8)>`

Comment (younies) [reply provider/datagen/src/transform/cldr/units/prefix.rs:19]: we can use the function as @robertbastian  suggest and get it to it back later

Comment (Manishearth) [reply provider/datagen/src/transform/cldr/units/prefix.rs:19]: Should be `parse_prefix(s: &str) -> Option<(u8, u8, usize)>` where the return type in `(base, power, index)`

**Comment by sffc** [comment]
This PR does not look finished yet, so I changed it to Draft. The PR is adding some new internal struct definitions but doesn't appear to use them anywhere.

**Pull request closed without merging**